src/components/pages/blog-post/blog-post.tsx

```diff
@@ -33,114 +56,246 @@ const PanelContainer = styled.div`
   }
 `;
 
-const contentfulRenderOptions: Options = {
-  renderMark: {
-    [MARKS.CODE]: (text) => <code className="language-text">{text}</code>,
-  },
-  renderNode: {
-    [INLINES.HYPERLINK]: (props, children) =>
-      customComponents.a({
-        children,
-        href: props.data.uri,
-        target: '_blank',
-        rel: 'nofollow noopener noreferrer',
+/**
+ * Custom Contentful Rich Text Renderer
+ *
+ * Will automatically filter out ContentfulFootnote elements and appends an
+ * ordered list of footnotes after the rendered rich text.
+ *
+ * @param data rich text data
+ * @returns rendered elements
+ */
+const customContentfulRenderer = (
+  data: RenderRichTextData<ContentfulRichTextGatsbyReference>,
+) => {
+  /**
+   * Filter out footnotes from content references
+   */
+  const footnotes: ContentfulFootnoteReference[] = data.references
+    .filter(
+      (reference) =>
+        reference.__typename === ContentfulCustomModel.CONTENTFUL_FOOTNOTE,
+    )
+    .map((reference) => reference as ContentfulFootnoteReference);
```

Comment:
You map just for the purpose of getting the types right? Can't you just type case to `ContentfulFootnoteReference[]` instead? The last mapping operation looks useless from a functional point ?